### PR TITLE
docs: add optimism chains

### PR DESCRIPTION
**Pull request by mattstam**

### CODE REVIEW

File: book/onchain-verification/contract-addresses.md [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
+| 84532    | Base Sepolia     | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://sepolia.basescan.org/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e)           |
+| 10       | Optimism         | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://optimistic.etherscan.io/address/0x3b6041173b80e77f038f3f2c0f9744f04837185e)        |
+| 11155420 | Optimism Sepolia | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](hhttps://sepolia-optimism.etherscan.io/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e) |
+| 534351   | Scroll Sepolia   | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://sepolia.scrollscan.com/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e)         |
+| 534352   | Scroll           | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://scrollscan.com/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e)                 |
+
+A complete reference for all of the contract addresses can be also be found in the [SP1 Contracts Repo](https://github.com/succinctlabs/sp1-contracts/blob/main/contracts/deployments).
```
Comment (ratankaliani) [inline book/onchain-verification/contract-addresses.md:19]: nit: A complete reference for all of the `SP1Verifier` contract addresses

File: book/onchain-verification/contract-addresses.md [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
+
+Whenever a new prover version is deployed, the gateway contract will be updated to support it with [addRoute()](https://github.com/succinctlabs/sp1-contracts/blob/main/contracts/src/ISP1VerifierGateway.sol#L65). In the case that a prover version has an issue, the gateway contract will stop supporting it with [freezeRoute()](https://github.com/succinctlabs/sp1-contracts/blob/main/contracts/src/ISP1VerifierGateway.sol#L71).
```
Comment (ratankaliani) [inline book/onchain-verification/contract-addresses.md:21]: nit:
Whenever a verifier for a new SP1 version is deployed,

If a verifier for an SP1 version has an issue, the route will be frozen with [freezeRoute()]

File: book/onchain-verification/contract-addresses.md [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
-| 534351   | Scroll Sepolia   | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://sepolia.scrollscan.com/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e) |
-| 534352   | Scroll           | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://scrollscan.com/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e)         |
-| 8453     | Base             | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://basescan.org/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e)           |
-| 84532    | Base Sepolia     | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://sepolia.basescan.org/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e)   |
-
-**Currently officially supported version of SP1 is v1.1.0.** If you'd like official support for a verifier on a different chain, please ask in the [SP1 Telegram](https://t.me/+AzG4ws-kD24yMGYx).
+When using SP1, we recommend using the deployed verifier gateways. On each supported chain, an [SP1VerifierGateway](https://github.com/succinctlabs/sp1-contracts/blob/main/contracts/src/ISP1VerifierGateway.sol) can automatically route your SP1 proof to the correct verifier based on the prover version. The full list can be found below:
```
Comment (ratankaliani) [inline book/onchain-verification/contract-addresses.md:3]: nit: To verify SP1 proofs on-chain, we recommend using our deployed verifier gateways. For the chains listed below, a [SP1VerifierGateway](https://github.com/succinctlabs/sp1-contracts/blob/main/contracts/src/ISP1VerifierGateway.sol) can automatically route your SP1 proof to the correct verifier based on the SP1 version.

File: book/onchain-verification/contract-addresses.md (hunk 1 above)
Comment (mattstam) [reply book/onchain-verification/contract-addresses.md:19]: fixed

File: book/onchain-verification/contract-addresses.md (hunk 2 above)
Comment (mattstam) [reply book/onchain-verification/contract-addresses.md:21]: fixed

File: book/onchain-verification/contract-addresses.md (hunk 3 above)
Comment (mattstam) [reply book/onchain-verification/contract-addresses.md]: fixed

File: book/onchain-verification/contract-addresses.md [hunk 4]
```diff
@@ -1,20 +1,24 @@
...
-| 534351   | Scroll Sepolia   | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://sepolia.scrollscan.com/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e) |
-| 534352   | Scroll           | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://scrollscan.com/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e)         |
-| 8453     | Base             | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://basescan.org/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e)           |
-| 84532    | Base Sepolia     | [0x3B6041173B80E77f038f3F2C0f9744f04837185e](https://sepolia.basescan.org/address/0x3B6041173B80E77f038f3F2C0f9744f04837185e)   |
-
-**Currently officially supported version of SP1 is v1.1.0.** If you'd like official support for a verifier on a different chain, please ask in the [SP1 Telegram](https://t.me/+AzG4ws-kD24yMGYx).
+To verify SP1 proofs on-chain, we recommend using our deployed verifier gateways. For the chains listed below, a [SP1VerifierGateway](https://github.com/succinctlabs/sp1-contracts/blob/main/contracts/src/ISP1VerifierGateway.sol) can automatically route your SP1 proof to the correct verifier based on the SP1 version.
```
Comment (ratankaliani) [inline book/onchain-verification/contract-addresses.md:3]: nit: an `SP1VerifierGateway`.

**Review by ratankaliani — APPROVED** [review]
LGTM, one small nit

File: book/onchain-verification/contract-addresses.md (hunk 4 above)
Comment (mattstam) [reply book/onchain-verification/contract-addresses.md]: fixed

**Pull request merged**